src/cascadia/TerminalSettingsEditor/Profiles.h

```diff
@@ -5,15 +5,66 @@
 
 #include "Profiles.g.h"
 #include "ProfilePageNavigationState.g.h"
+#include "ProfileViewModel.g.h"
 #include "Utils.h"
+#include "ViewModelHelpers.h"
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
+    struct ProfileViewModel : ProfileViewModelT<ProfileViewModel>, ViewModelHelper<ProfileViewModel>
+    {
+    public:
+        ProfileViewModel(const Model::Profile& profile) :
+            _profile{ profile } {}
+
+        PERMANENT_OBSERVABLE_PROJECTED_SETTING(_profile, Guid);
+        PERMANENT_OBSERVABLE_PROJECTED_SETTING(_profile, ConnectionType);
+        OBSERVABLE_PROJECTED_SETTING(_profile, Name);
```

Comment:
as happy as these new macros make me, I friggin hate that there's _another_ place we'll need to make sure to add the setting to. Chalk this up as another reason to have `AddATerminalSetting.md`

Reply:
I know, I know.

Reply:
I'll actually go ahead and add `AddATerminalSetting.md` to my follow-up tasks.